### PR TITLE
also check opencl headers

### LINKED ISSUE #919: [build.sh] Error to run build.sh
Hi, I downloaded the code and followed the steps but I have an error.

Can Anybody help me?

I have Ubuntu 18.04 x64

imagen

This error was produced when I executed CC=clang-6.0 CXX=clang++-6.0 INSTALL_PREFIX=~/.local ./build.sh

### PR DESCRIPTION
**Pull request by borg323**
This should fix #919

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -311,7 +311,7 @@ if get_option('build_backends')
       deps += [ opencl_framework ]
       has_opencl = true
 
-  elif opencl_lib.found()
+  elif opencl_lib.found() and cc.has_header('CL/opencl.h', args: '-I' + get_option('opencl_include')[0])
```
Comment (ddobbelaere) [inline meson.build]: Maybe make it also resilient to future additions of include directories, in the spirit of https://github.com/LeelaChessZero/lc0/blob/06748a4a35b297164e36400a576bc64031fa0415/meson.build#L222
Just an idea, I imagine someone editing `meson_options.txt` later won't recall or even know that this check only considers the first include directory.

Comment (borg323) [reply meson.build]: We use the same trick in <https://github.com/LeelaChessZero/lc0/blob/master/meson.build#L160>. 
Maybe we should change the option type from `array` to `string` when only a single argument is supported.

Comment (borg323) [reply meson.build]: I did make `opencl_include` a string option with the latest commit. There are a few more candidates but I want to consider them separately

File: meson_options.txt [hunk 2]
```diff
@@ -14,8 +14,8 @@ option('openblas_include',
        description: 'Paths to openblas include directories')
 
 option('opencl_include',
-       type: 'array',
-       value: ['/usr/include/'],
+       type: 'string',
+       value: '/usr/include/',
        description: 'Paths to OpenCL include directories')
```
Comment (ddobbelaere) [inline meson_options.txt]: This is really nitpicking, but now it's "Path" :)

Comment (borg323) [reply meson_options.txt]: If only every review comment was this easy..

Comment (borg323) [reply meson_options.txt]: ...and if only I was more careful when writing patches!

**Pull request merged**